src/TwigExtension.php

```diff
@@ -274,6 +274,20 @@ public function bibliography(array $value, string $locale, array $styles = []):
     else {
       $bibliography = $render->bibliography(null, $styles, $json_data);
     }
-    return $bibliography;
+
+    $render_bibliography = [
+      '#type' => 'container',
+      '#attributes' => [
+        'id' => 'bibliography' . 'test-unique-id',
+        'class' => ['bibliography'],
+      ],
+      '#attached' => [
+        'library' => 'format_strawberryfield/citations_strawberry'],
+    ];
+    $render_bibliography['bibliography'] = [
+      '#markup' => \Drupal\Core\Render\Markup::create($bibliography),
+    ];
+    $rendered_bibliography = \Drupal::service('renderer')->render($render_bibliography);
```

Comment:
@aksm this is great!!. Hint. Try returning it without the -> render and check it out. Twig should be able to render Render arrays directly. If that does not work out let me know!

Reply:
@DiegoPino Tried that at one point and got the `Return value must be of type string` error. It looks like that only works with functions, not filters? Am I missing something?